main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strconv"
+
+	"github.com/ONSdigital/dp-reporter-client/reporter"
+	"github.com/ONSdigital/dp-search-builder/config"
+	"github.com/ONSdigital/dp-search-builder/elasticsearch"
+	"github.com/ONSdigital/dp-search-builder/service"
+	"github.com/ONSdigital/go-ns/kafka"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/rchttp"
+)
+
+func main() {
+	log.Namespace = "dp-search-builder"
+
+	cfg, err := config.Get()
+	if err != nil {
+		log.Error(err, nil)
+		os.Exit(1)
+	}
+
+	envMax, err := strconv.ParseInt(cfg.KafkaMaxBytes, 10, 32)
+	if err != nil {
+		log.ErrorC("encountered error parsing kafka max bytes", err, nil)
+		os.Exit(1)
+	}
+
+	syncConsumerGroup, err := kafka.NewSyncConsumer(cfg.Brokers, cfg.ConsumerTopic, cfg.ConsumerGroup, kafka.OffsetOldest)
```

Comment:
Using offsetOldest is good, but make sure that handling messages is idempotent. i.e. handling the same message twice will not break things

Reply:
If a message is replayed it will delete the current index and build it again, but with the same data (unless the data has changed in the other datastores), so it is idempotent. Only issue is the time that exists between deleting an index and rebuilding it.

Reply:
👍